back/src/Controller/IoSocketController.ts

```diff
@@ -1,36 +1,80 @@
 import socketIO = require('socket.io');
 import {Socket} from "socket.io";
 import * as http from "http";
-import {MessageUserPosition} from "@Model/Websocket/MessageUserPosition";
+import {MessageUserPosition} from "../Model/Websocket/MessageUserPosition"; //TODO fix import by "_Model/.."
+import {ExSocketInterface} from "../Model/Websocket/ExSocketInterface"; //TODO fix import by "_Model/.."
+import Jwt, {JsonWebTokenError} from "jsonwebtoken";
+import {SECRET_KEY} from "../Enum/EnvironmentVariable"; //TODO fix import by "_Enum/..."
 
 export class IoSocketController{
     Io: socketIO.Server;
     constructor(server : http.Server) {
         this.Io = socketIO(server);
+
+        //authentication with token. it will be decodes and stock in socket.
+        this.Io.use( (socket: Socket, next) => {
+            if (!socket.handshake.query || !socket.handshake.query.token) {
+                return next(new Error('Authentication error'));
+            }
+            Jwt.verify(socket.handshake.query.token, SECRET_KEY, (err: JsonWebTokenError, tokenDecoded: object) => {
+                if (err) {
+                    return next(new Error('Authentication error'));
+                }
+                (socket as ExSocketInterface).token = tokenDecoded;
+                next();
+            });
+        });
+
         this.ioConnection();
     }
 
     ioConnection() {
         this.Io.on('connection',  (socket: Socket) => {
-            //TODO check token access
-
             /*join-rom event permit to join one room.
                 message :
                     userId : user identification
                     roomId: room identification
-                    positionXUser: user x position map
-                    positionYUser: user y position on map
+                    position: position of user in map
+                        x: user x position on map
+                        y: user y position on map
             */
-            socket.on('join-room', (message : MessageUserPosition) => {
-                socket.join(message.roomId);
+
+            socket.on('join-room', (message : string) => {
```

Comment:
For the record, the message send by "emit" and received by "on" does not have to be a string.
It can be JSON directly.

https://socket.io/docs/server-api/#socket-emit-eventName-%E2%80%A6args-ack

But we can keep it this way for now. It is likely that we will want to use something more performant than JSON to share positions (since this will be sent a lot).
Something like protobuf or Apache Thrift.

But we can keep that for a next version.